tests/json_test/json_test.cpp

```diff
@@ -7086,6 +7086,48 @@ suite expected_tests = [] {
    };
 };
 
+struct custom_struct
+{
+   std::string str{};
+};
+
+namespace glz::detail
+{
+   template <>
+   struct from_json<custom_struct>
+   {
+      template <auto Opts>
+      static void op(custom_struct& value, auto&&... args)
+      {
+         read<json>::op<Opts>(value.str, args...);
+         value.str += "read";
+      }
+   };
+
+   template <>
+   struct to_json<custom_struct>
+   {
+      template <auto Opts>
+      static void op(custom_struct& value, auto&&... args) noexcept
+      {
+         value.str += "write";
+         write<json>::op<Opts>(value.str, args...);
+      }
+   };
+}
+
+suite custom_struct_tests = [] {
+   "custom_struct"_test = [] {
+      custom_struct obj{};
+      std::string s{};
```

Comment:
try changing it to const

`std::string const s{};`

Reply:
You can't use const for your buffer if you are writing, because it may need to grow the buffer.

Reply:
my bad, I was thinking about custom_struct obj{};
But I am busy at work and I made mistake

Reply:
No problem